api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineWithPermissionsTest.java

```diff
@@ -71,80 +67,264 @@
     private Folder folder2;
     private Pipeline pipeline1;
     private Pipeline pipeline2;
-    private AclTestDao.AclSid testUserSid1;
-    private AclTestDao.AclSid testUserSid2;
 
     @Before
     public void setUp() {
-        AclTestDao.AclClass folderAclClass = new AclTestDao.AclClass(Folder.class.getCanonicalName());
-        aclTestDao.createAclClassIfNotPresent(folderAclClass);
-
-        AclTestDao.AclClass pipelineAclClass = new AclTestDao.AclClass(Pipeline.class.getCanonicalName());
-        aclTestDao.createAclClassIfNotPresent(pipelineAclClass);
-
-        testUserSid1 = new AclTestDao.AclSid(true, TEST_OWNER1);
-        aclTestDao.createAclSid(testUserSid1);
-
-        testUserSid2 = new AclTestDao.AclSid(true, TEST_OWNER2);
-        aclTestDao.createAclSid(testUserSid2);
-
         folder1 = getFolder(TEST_FOLDER1, TEST_OWNER1);
-        folderDao.createFolder(folder1);
+        folder1.setId(1L);
 
         folder2 = getFolder(TEST_FOLDER2, TEST_OWNER2);
-        folder2.setParentId(folder1.getId());
-        folderDao.createFolder(folder2);
+        folder2.setId(2L);
 
         pipeline1 = constructPipeline(TEST_PIPELINE1, TEST_PIPELINE_REPO, TEST_PIPELINE_REPO_SSH, folder2.getId());
         pipeline1.setOwner(TEST_OWNER1);
-        pipelineDao.createPipeline(pipeline1);
+        pipeline1.setId(1L);
 
         pipeline2 = constructPipeline(TEST_PIPELINE2, TEST_PIPELINE_REPO, TEST_PIPELINE_REPO_SSH, folder1.getId());
         pipeline2.setOwner(TEST_OWNER1);
-        pipelineDao.createPipeline(pipeline2);
+        pipeline2.setId(2L);
+
+        pipeline1.setParentFolderId(folder2.getId());
+        pipeline2.setParentFolderId(folder1.getId());
+        pipeline1.setParent(folder2);
+        pipeline2.setParent(folder1);
+
+        folder1.setPipelines(Collections.singletonList(pipeline2));
+        folder2.setPipelines(Collections.singletonList(pipeline1));
     }
 
     @Test
-    @Transactional(propagation = Propagation.REQUIRES_NEW)
-    @WithMockUser(username = TEST_OWNER1)
-    public void testMergePermissions() {
+    public void shouldReturnPermissionsWhenGrantedToPipelineCase1() {
         // key - pipelineId, value - <sid, mask>
-        Map<Long, Map<String, Integer>> expectedMap = new HashMap<>();
-        HashMap<String, Integer> permissionsForPipeline1 = new HashMap<>();
-        HashMap<String, Integer> permissionsForPipeline2 = new HashMap<>();
+        final Map<Long, Map<String, Integer>> expectedMap = new HashMap<>();
+        final Map<String, Integer> permissionsForPipeline1 = new HashMap<>();
+        final Map<String, Integer> permissionsForPipeline2 = new HashMap<>();
```

Comment:
I suggest moving these declarations closer to the call site.

Reply:
I thought that we should just move these declarations to the end of each test. Something like this 

```java
final Map<String, Integer> permissionsForPipeline1 = Collections.singletonMap(TEST_USER_1.toUpperCase(), 5);
final Map<String, Integer> permissionsForPipeline2 = Collections.singletonMap(TEST_USER_1.toUpperCase(), 1);
final Map<Long, Map<String, Integer>> expectedMap = new HashMap<>();
expectedMap.put(pipeline1.getId(), permissionsForPipeline1);
expectedMap.put(pipeline2.getId(), permissionsForPipeline2);
assertPipelineWithPermissions(expectedMap);
```

Reply:
In the first test it will work, but in the other tests it will be replaced existing map in the expected map and tests will not passed.